addons/sale_gift_card/models/sale_order.py

```diff
@@ -50,6 +50,11 @@ def _pay_with_gift_card(self, gift_card):
             })
         return error
 
+    def _get_gift_card_user(self):
+        if self.website_id:
+            return self.env.user.partner_id
```

Comment:
Instead of adding 2 return , maybe simply do a return self.env.user.partner_id if self.website_id else self.partner_id
+ Could you a